WPMedia/fusion-news-theme-blocks
# add fix for redirect

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description

- This is a fix for the log in redirect. When logged in the user would not get redirected back to the page they were on.
- This also has a fix for the RegWall button getting blocked in mobile.

## Jira Ticket

- [ASUB-8279](https://arcpublishing.atlassian.net/browse/ASUB-8279)
- [ASUB-8199](https://arcpublishing.atlassian.net/browse/ASUB-8199)

## Acceptance Criteria

- Should redirect to the previous page after logging in.
- Registration Wall should be above the fold on Safari and Chrome

## Test Steps
Log in Redirect:
- Sign up and make sure you have an account with the website. Create a sign up page if needed
- Create a page for the login page to redirect to with a link to the log in page that you can click on.
- Create a log in page if you do not have one.
- Open the page for the redirect page.
- Click on the log in link - this will redirect you to the log in page.
- Fill out the form to log in.
- Log in should redirect you to the redirect page you created.

RegWall button:
- Create a page with the Paywall block
- Add the RegWall
- You will need to comment some code and return only the RegWall in the paywall theme block. This will allow you to see the RegWall without any logic to trigger a showing of the RegWall.
```
	return (
		<RegwallOffer
			actionText={registerActionText}
			actionUrl={registerActionUrl}
			displayMode={displayMode}
			headlineText={registerHeaderText}
			linkPrompt={payLinkPrompt}
			linkText={linkText}
			linkUrl={linkUrl}
			reasonPrompt={registerReasonPrompt}
			subheadlineText={registerSubHeaderText}
			className={BLOCK_CLASS_NAME}
		/>
	);
```
- You will need to either open the Paywall page on a iPhone device using the IP address or you can open it in the Xcode simulator. You can only test Chrome on a mobile device.

## Effect Of Changes

### Before

Login Redirect:
- Click log in from an article.
- Log in - then it would send you to the home page and not the article you were on.

RegWall:
<img width="409" alt="Screenshot 2024-03-14 at 7 28 18 PM" src="https://github.com/WPMedia/arc-themes-blocks/assets/5377648/b1ff9f68-d2ee-4913-b0ed-2263d3fa307f">

### After
Login Redirect:
- Should redirect you to the article page you were on after you log in.

RegWall:
<img width="412" alt="Screenshot 2024-03-14 at 7 20 21 PM" src="https://github.com/WPMedia/arc-themes-blocks/assets/5377648/452a18da-23e2-42f8-97aa-d8fc1a168f81">

![IMG_60E0869E7B74-1](https://github.com/WPMedia/arc-themes-blocks/assets/5377648/04edeb3b-84ef-4e78-9725-87cab52d93dd)
## Dependencies or Side Effects

None

- Manifest pr for new block:
- Feature pack pr for local development:
- How to deploy to news: https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138682964/News+Theme+2.0+Migration+Development+Deployment+Notes
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Dependency on another PR that needs to be merged first

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[ASUB-8279]: https://arcpublishing.atlassian.net/browse/ASUB-8279?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ASUB-8199]: https://arcpublishing.atlassian.net/browse/ASUB-8199?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
